crates/evm/src/query.rs

```diff
@@ -35,14 +35,30 @@ use crate::rpc_helpers::*;
 use crate::{BloomFilter, EthResult, Evm, EvmChainConfig, FilterBlockOption, FilterError};
 
 // Gas per transaction not creating a contract.
-const MIN_TRANSACTION_GAS: u64 = 21_000u64;
+pub(crate) const MIN_TRANSACTION_GAS: u64 = 21_000u64;
 
 /// https://github.com/paradigmxyz/reth/pull/7133/files
 /// Allowed error ratio for gas estimation
 /// Taken from Geth's implementation in order to pass the hive tests
 /// <https://github.com/ethereum/go-ethereum/blob/a5a4fa7032bb248f5a7c40f4e8df2b131c4186a4/internal/ethapi/api.go#L56>
 const ESTIMATE_GAS_ERROR_RATIO: f64 = 0.015;
 
+/// The result of gas estimation.
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub(crate) struct EstimatedGas {
```

Comment:
since it's not l1_fee but l1_fee / base_fee now, let's call it l1_fee_overhead or smth like that